docs/poa/flows.md

```diff
@@ -0,0 +1,107 @@
+# Flows
+
+## PoA Primary Production Flow
+When the node is configured with a POA key, produce blocks and notify network.
+
+```mermaid
+sequenceDiagram
+    participant POA as PoA Service
+    participant BP as Block Producer
+    participant TX as Transaction Pool
+    participant R as Relayer
+    participant E as Executor
+    participant D as Database
+    participant P2P as P2P
+    
+    POA->>BP: produce block
+    BP->>TX: select_txs
+    TX-->>BP: 
+    BP->>R: get_da_height
+    R-->>BP: 
+    BP->>E: execute and store
+    E->>D: save diff
+    E-->>BP: 
+    BP-->>POA: Malleated block
+    POA->>R: get_poa_key
+    R-->>POA: 
+    POA->>POA: sign block
+    POA->>D: insert_consensus_data
+    POA->>P2P: broadcast_new_block
+```
+
+## PoA Synchronization Flow
+
+When a node is behind peers, download the block data and catch up.
+
+```mermaid
+sequenceDiagram
+    participant S as Synchronizer
+    participant P2P as P2P
+    participant POA as PoA Service
+    participant BI as Block Importer
+    participant R as Relayer
+    participant E as Executor
+    participant D as Database
+    participant TX as Transaction Pool
+    
+    S->>+P2P: get_peer_heights
+    P2P-->>-S: 
+    alt is not behind peers
+    S-->>S: sleep
+    end
+    S->>+P2P: get_missing_blocks
+    P2P-->>-S: 
+    S->>+POA: verify block signatures
+    POA->>+R: await_synced + get_da_height
+    R-->>-POA: 
+    POA-->>-S: 
+    S->>+BI: commit
+    BI->>+R: check_da_height
+    R-->>-BI: 
+    BI->>+E: validate_and_store 
+    E->>+D: save diff
+    D-->>-E: 
+    E-->>-BI: 
+    BI->>+D: insert_consensus_data
+    D-->>BI: 
+    BI->>+TX: drop_committed_txs
+    TX-->>-BI: 
+    BI-->>-S: 
+    
+```
+
+## PoA Gossip-Sync Flow
+
+When a non-producer is synced to the PoA node, bypass Synchronizer and react to block broadcasts.
```

Comment:
My thinking was that sync would be a polling mechanism for already sealed blocks, whereas gossipsync via the CM would be push-based. 

The motivation for this because when we move to PoS, it will then require additional back and forth between synchronizer and CM to convert an unsealed block into a sealed block, pending enough votes. Since block broadcast messages are very specific to a consensus algorithm, I think it will be easiest to let the CM decide on what to do with them.

The downside is if the synchronizer starts polling for the same block the CM is already in the process of sealing/finalizing. This could be mitigated via some global "is syncing" toggle switch like ethereum and most other chains use.

Reply:
P2P forwards all consensus information to a separate channel/stream to CM. 
- If you are a producer, you collect votes. After the collection of votes for the block, you will commit it. After committing, you will broadcast it to the network.
- If you are not a producer in the PoS but part of a council, you verify the block and send a vote to P2P. You ignore votes.
- If you are not a producer or part of the council, you ignore all information.

FS of each node will receive broadcasted block and will handle synchronization=) 

As we discussed, we can always cache the state of the executed pending block, to speed up the commit. But we don't need to collect votes in the CM(it is possible in the future as part of optimization).

Reply:
Updated to use the synchronizer for finalized block gossip